packages/frameworks/vue/src/factory.tsx

```diff
@@ -65,7 +65,7 @@ export function withAsChild(__component: RenderFunctionArgs) {
     },
     setup(props, { attrs, slots }) {
       const instance = getCurrentInstance()
-      if (!props.asChild) return () => <__component {...attrs}>{slots.default?.()}</__component>
+      if (!props.asChild) return () => h(__component, { ...attrs }, slots.default?.())
```

Comment:
Why this change?

Reply:
fix a type error
<img width="898" alt="image" src="https://github.com/chakra-ui/ark/assets/36906371/78de7ed3-088d-4669-9b2c-c97f9998f773">

Reply:
Okay, it works and typecheck is good, so I approve this